src/React/Promise/Deferred.php

```diff
@@ -96,6 +122,35 @@ public function resolver()
         return $this->resolver;
     }
 
+    public function cancel()
+    {
+        if (null === $this->canceller || null !== $this->completed) {
+            return;
+        }
+
+        $canceller = $this->canceller;
+        $this->canceller = null;
+
```

Comment:
For consistency, you should check here for `is_callabe($canceller)` and trigger a notice when `false` like done [here](https://github.com/reactphp/promise/blob/1.0/src/React/Promise/FulfilledPromise.php#L19-L23), [here](https://github.com/reactphp/promise/blob/1.0/src/React/Promise/RejectedPromise.php#L17-L20) and [here](https://github.com/reactphp/promise/blob/1.0/src/React/Promise/Deferred.php#L30-L32).